### DISCUSSION TITLE
CollectionViewのCell内に、画面を長押ししたら画像が表示されるようにしたい

### DISCUSSION THREAD

**Discussion by mahirooo** [lit-iphone-course/lit-iphone-course#31] [Question-original] [answered]
**(1) 実現したいこと**
PushViewControllerにおいて、画面を長押しした時に、Cell内に画像"stampBlue"が１つずつ表示されてほしい

**(2) 実現するために自分が試した内容とその結果**
cell一つ一つに画像が表示されるようにコードを書いた
→PushViewControllerに遷移した瞬間に落ちるようになった

**(3) 調べた内容や資料**
https://sunnyday-travel-aso-6487.ssl-lolipop.jp/programing/swift/collectionview-imag/
https://i-app-tec.com/ios/collectionview.html

**(4) 原因だと思われそうな箇所(ソースコードをコピペして貼る)**
https://github.com/mahirooo/TampStamp/blob/bdac06f2410cb0af902074486207ab284da05106/TampStamp/PushViewController.swift#L52-L57

**(5) GitHubのリンク**
https://github.com/mahirooo/TampStamp

**Comment by txxxxc** [comment]
深夜にごめん！
> PushViewControllerに遷移した瞬間に落ちるようになった
これで出たエラーの詳細を教えてもらえるとありがたいです！

**Comment by txxxxc** [comment, marked as answer]
ビルドして試してみました！多分原因わかったので共有します！
原因としては配列の存在しないindexにアクセスしているところです！
具体的には`numberOfItemInSection`でセルの数を8にしているのに対して空の`stampImage`にアクセスしている部分です！

該当箇所↓
```swift
var stampImage = [String]()

// 中略
func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
  return 8
}

func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
      // 中略
      // stampImageの2番目の要素とか無いです；；って怒られている
      let cellImage = UIImage(named: stampImage[indexPath.row])
}

```

具体的には、`stampImage = ["", "", "", "", "", "", "", "",]`って感じで予め空の文字列を突っ込んであげるといけると思います！
試してみてくださいー！

**Reply by txxxxc** [reply]
このエラー解決できそうですか...?
自分の回答が的はずれだったら全然言ってもらえると嬉しいですー！🙇‍♂️

**Reply by mahirooo** [reply]
ああああごめんなさい！ちょっとだけ仕様変更したので今から試してみます！！

**Reply by mahirooo** [reply]
こちら解決できたのですがまた別の問題が出てきたので、別で質問しました！！ありがとうございます！！！
